manifest/provider/apply.go

```diff
@@ -165,7 +165,12 @@ func (s *RawProviderServer) ApplyResourceChange(ctx context.Context, req *tfprot
 
 		gvk, err := GVKFromTftypesObject(&obj, m)
 		if err != nil {
-			return resp, fmt.Errorf("failed to determine resource GVK: %s", err)
+			resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+				Severity: tfprotov5.DiagnosticSeverityError,
+				Summary:  "Failed to determine resource GVK",
```

Comment:
Same with this one, `GVK` is kind of an obscure acronym.

Perhaps here we could say something like 
```
Summary: "Failed to determine the kind for the resource.",
Detail: `This can happen when the "apiVersion" or "kind" fields are not present in the manifest, or when the corresponding "kind" or "apiVersion" could not be found on the Kubernetes cluster`
```